python/pyarrow/src/arrow_to_pandas.cc

```diff
@@ -155,7 +156,7 @@ struct WrapBytes<FixedSizeBinaryType> {
 };
 
 static inline bool ListTypeSupported(const DataType& type) {
-  switch (type.id()) {
+  switch (type.storage_id()) {
```

Comment:
Instead of using storage_id here, I think you can add a `case Type::EXTENSION` which then recursively calls ListTypeSupported again on the storage type (similarly how it is done for list types)